server/src/catalog.rs

```diff
@@ -65,223 +112,481 @@ impl Catalog {
-                (1, "dk.aau.modelardb.core.models.UncompressedModelType") => (),
-                (2, "dk.aau.modelardb.core.models.PMC_MeanModelType") => (),
-                (3, "dk.aau.modelardb.core.models.SwingFilterModelType") => (),
-                (4, "dk.aau.modelardb.core.models.FacebookGorillaModelType") => (),
-                _ => return Err(ParquetError::General("unsupported model type".to_string())),
-            }
+// TODO: update after implementing the compression component.
+/// Metadata required to query a model table.
+#[derive(Debug)]
+pub struct ModelTableMetadata {
+    /// Name of the model table.
+    pub name: String,
+    /// Location of the model table's segment folder in an `ObjectStore`.
+    pub segment_path: ObjectStorePath,
+    // TODO: remove segment_file_legacy_schema when refactoring query engine.
+    /// Schema of the Apache Parquet files used by the [JVM-based version of
+    /// ModelarDB] for storing segments.
+    ///
+    /// [JVM-based version of ModelarDB]: https://github.com/modelardata/ModelarDB
+    pub segment_file_legacy_schema: SchemaRef,
+    /// Cache that maps from time series id to sampling interval for all time
+    /// series in the table.
+    pub sampling_intervals: Int32Array,
+    // TODO: rename dimension members to tags when refactoring query engine.
+    /// Cache that maps from time series id to data warehouse dimension members
+    /// for all time series in the table. Each `Array` in
+    /// `denormalized_dimensions` is a level in a dimension.
+    pub denormalized_dimensions: Vec<ArrayRef>,
+}
+
+impl ModelTableMetadata {
+    /// Read and check the metadata for a model table produced by the [JVM-based
+    /// version of ModelarDB]. Return `ParquetError` if the metadata cannot be
+    /// read or if the model table uses unsupported model types.
+    ///
+    /// [JVM-based version of ModelarDB]: https://github.com/modelardata/ModelarDB
+    fn try_new(
+        folder_name: String,
+        folder_path: String,
+        segment_file_legacy_schema: &SchemaRef,
+    ) -> Result<Arc<Self>, ParquetError> {
+        // Pre-compute the path to the segment folder in the object store.
+        let segment_path = Self::create_object_store_path_to_segment_folder(&folder_path)?;
+
+        // Ensure only supported model types are used.
+        Self::check_model_type_metadata(&folder_path)?;
+
+        // Read time series metadata. The sampling intervals and dimension
+        // members are shifted by one as the time series ids start at one.
+        let time_series_file = folder_path.clone() + "/time_series.parquet";
+        let path = Path::new(&time_series_file);
+        if let Ok(file) = File::open(&path) {
+            let rows = Self::read_entire_apache_parquet_file(&folder_name, file)?;
+            let sampling_intervals = Self::extract_and_shift_int32_array(&rows, 2)?;
+            let denormalized_dimensions = // Columns with metadata as strings.
```

Comment:
I agree with the [Comments Section in the Rust Style Guide](https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/guide.md#comments) that comments that relate to a block of code generally should be on their own line:

_Prefer to put a comment on its own line. Where a comment follows code, there should be a single space before it_. 

However, since this comments was specifically added to explain `denormalized_dimensions`, and the style guide does not directly prohibit having code and comments on the same line, it made sense to me to put it on the same line. Although maybe I have just been [writing](https://www.gnu.org/software/emacs/manual/html_node/elisp/Comment-Tips.html) too much [Lisp](https://books.google.dk/books?id=CIYUAwAAQBAJ&pg=PA150&lpg=PA150), so we can also move the comment to the line above if you think that is better?

Reply:
I am not against inline comments in general. In this case it is just a little weird since 
`let denormalized_dimensions = Self::extract_and_shift_denormalized_dimensions(&rows, 4)?;` is one line of code and the comment is technically in the middle of this line, while most inline comments would be after the line. But if you prefer it like this then it is completely fine.

Reply:
Ok, I generally think of each line in the document as separate, but I can see why the comment looks strange if you think of line 293-294 as a single line of code. I will try to take that into account if I use inline comments in the future.